core/drand_public.go

```diff
@@ -129,7 +131,9 @@ func (d *Drand) PublicRandStream(req *drand.PublicRandRequest, stream drand.Publ
 		var err error
 		b.Store().Cursor(func(c beacon.Cursor) {
 			for bb := c.Seek(req.GetRound()); bb != nil; bb = c.Next() {
-				if err = stream.Send(beaconToProto(bb)); err != nil {
+				protoBeacon := beaconToProto(bb)
+				protoBeacon.Timestamp = beacon.TimeOfRound(d.group.Period, d.group.GenesisTime, bb.Round)
```

Comment:
This will lie if the node failed for a while and is catching up, right? we'd ideally like to have the actual time stamp that a round becomes available, to compare that with when the round was scheduled to occur

Reply:
I asked that question on slack and @hsanjuan replied on the expected round time. This information is verifiable by any endpoint tbh but @hsanjuan wanted to see it on the output. Similar to `Randomness` I'm fine to provide it for debug / ease-of-use.

This question is a bigger question of the semantics of timestamp:
- timestamp of rounds: this is a fixed public mapping given a period, genesis and round time
- timestamp of creation: this is a timestamp individual to each drand node, each drand node can put whatever they want.

I'm fine to return the first one, since it's deterministic. I'm much less fine to return the second one because Im afraid client would by mistake rely on it and there is no way to compute/verify that information.